source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,724 @@
+`pod network add-on`_.
 
-  If the result of the above commands is blank, you may need to add a ``profile`` to the :file:`/etc` directory. To do so, follow these steps.
+Some pod network add-ons enable advanced functionality with physical networks
+or cloud provider networks.
 
-  #. Create a `profile` in :file:`/etc`
+What is a Container Runtime Interface (CRI)?
+============================================
 
-     .. code-block:: bash
+Container runtimes are the underlying fabric that pod workloads execute inside
+of. Different container runtimes offer different balances between features,
+performance, and security. 
 
-        sudo touch profile
+Kubernetes allows integration various container runtimes via a container
+runtime interface (CRI). 
 
-  #. With your preferred editor, open `profile`, and enter your proxy settings. An example is shown below.
+What is CRI+O?
+--------------
 
-     .. code-block:: bash
+`CRI+O <https://cri-o.io/>`_ is a lightweight alternative to using Docker as
+the runtime for kubernetes. It allows Kubernetes to use any OCI-compliant
+runtime as the container runtime for running pods, such as runc and
+Kata Containers as the container runtimes.
 
-        export "HTTP_PROXY=http://proxy.example.com:443"
-        export "HTTPS_PROXY=http://proxy.example.com:445"
-        export "SOCKS_PROXY=http://proxy.example.com:1080"
-        export "NO_PROXY= site.com,.site.com,localhost,127.0.0.1,<master IP>
+CRI+O allows setting a different runtime per-pod.
 
-     .. note::
+What is containerd?
+-------------------
 
-        <master IP> can be obtained by running :command:`ifconfig`.
+`containerd <https://containerd.io/>`_ is the runtime that the Docker engine
+is built on top of. 
 
-  #. Save and exit the `profile`.
+Kubernetes can use containerd directly instead of going through the Docker
+engine for increased robustness and performance. See the `blog post on
+kubernetes containerd integration
+<https://kubernetes.io/blog/2018/05/24/kubernetes-containerd-integration-goes-ga/>`_
+for more details.
 
-  #. Update your system's environment settings by executing the following command:
+containerd allows setting a different runtime per-pod.
 
-     .. code-block:: bash
+What is Docker?
+---------------
 
-        sudo source profile
+`Docker <https://www.docker.com/>`_ is an engine for running software packaged
+as functionally complete units, called containers, using the same operating
+system kernel.
 
-  #. To ensure your system isn't running previous session variables, run:
+The default built-in runtime provided by Kubernetes is using the system Docker
+installation via Dockershim and as a result is one of the simplest to use. One
+limitation of using Dockershim is that all pods on the Kubernetes node will
+inherit and use the default runtime that Docker is set to use. To be able to
+specify a container runtime per-Kerbernetes service, use CRI+O or containerd. 
 
-     .. code-block:: bash
+What is Kata Containers?
```

Comment:
This had a different bold to the heading as compared to the other "What is ..."

Reply:
This is intentional. The 3 CRIs are a a subsection of the `What is a Container Runtime Interface (CRI)?`.

The `What is Kata Containers?` is not in the same level as the CRIs.

Reply:
Ah ok.  It is hard to read without all the proper 01.org formatting that would go around it.

Reply:
You should pull the fork and build it locally :)